src/scrapers/generic_social.iced

```diff
@@ -0,0 +1,80 @@
+{BaseScraper} = require './base'
+{constants} = require '../constants'
+{v_codes} = constants
+
+exports.GenericSocialScraper = class GenericSocialScraper extends BaseScraper
+  constructor : (opts) ->
+    super opts
+
+  # ---------------------------------------------------------------------------
+
+  _check_args : (args) ->
+    if not(args.username?)
+      new Error "Bad args to GenericSocialScraper proof: no username given"
+    else
+      null
+
+  # ---------------------------------------------------------------------------
+
+  hunt2 : ({username, name}, cb) ->
+    out = {}
+    bad_args_err = (e, rc) ->
+      out.rc = if rc? then rc else v_codes.BAD_ARGS
+      new Error "Bad args to GenericSocialScraper proofs" + if e then ": #{e}" else ""
+
+    if not(username?)
+      err = bad_args_err "no username given"
+    else if not(name?)
+      err = bad_args_err "no service name given"
+    else if name.indexOf('.') is -1
+      err = bad_args_err "service name `#{name}` is likely wrong - no dot in name."
+    else if not (pproof = @libs.create_paramproofs_url_and_path({ name, username }))?
```

Comment:
@mlsteele can you take a look? it doesn't pass `get_service_obj` (exposing full `ParamProofService` class) anymore